projects/plugins/boost/app/lib/critical-css/data-sync-actions/Set_Provider_Errors.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Lib\Critical_CSS\Data_Sync_Actions;
+
+use Automattic\Jetpack\WP_JS_Data_Sync\Contracts\Data_Sync_Action;
+use Automattic\Jetpack_Boost\Lib\Critical_CSS\Critical_CSS_State;
+
+/**
+ * Critical CSS Action: Store errors for a provider.
+ */
+class Set_Provider_Errors implements Data_Sync_Action {
+	/**
+	 * Handles the action logic.
+	 *
+	 * @param mixed            $data     JSON Data passed to the action.
+	 * @param \WP_REST_Request $_request The request object.
+	 */
+	public function handle( $data, $_request ) {
+		$state = new Critical_CSS_State();
+
+		if ( empty( $data['key'] ) || empty( $data['errors'] ) ) {
+			return array(
+				'success' => false,
+				'state'   => $state->get(),
+				'error'   => 'Invalid data',
+			);
+		}
+
+		$provider_key = sanitize_key( $data['key'] );
+		$errors       = $data['errors'];
```

Comment:
Is it not thoroughly checked on the way in automatically by DataSync?

Reply:
Oh dear. I just checked and it is not. Ok, fair point. I'll look into it.

Reply:
Ok as of ad70af1f64, DataSync actions now take a schema to validate incoming requests on registration. You can now know at this point in the code that `$data['errors']` is the right shape.

I think it may be debatable whether the schema should be supplied to the register function, or whether the Handler class should return a schema (so the schema is in code next to the handler), but for now I have done this. We can refactor after.

Reply:
I've noticed that the datasync schema types all throw `\Error` when they should throw `\Exception` so they're not being caught by the datasync handlers. But that is outside the scope of this PR. I'll submit another one to deal with that.